File: Java/opentraceability/src/main/java/opentraceability/gdst/events/GDSTTransshipmentEvent.java
package opentraceability.gdst.events;

import opentraceability.Constants;
import opentraceability.models.events.EventAction;
import opentraceability.models.events.ObjectEventBase;
import opentraceability.models.identifiers.PGLN;
import opentraceability.utility.attributes.OpenTraceabilityAttribute;
import opentraceability.utility.attributes.OpenTraceabilityJsonAttribute;

import java.net.URI;

public class GDSTTransshipmentEvent extends ObjectEventBase<GDSTILMD> implements IGDSTEvent {
    @OpenTraceabilityAttribute(ns = Constants.GDST_NAMESPACE, name = "productOwner")
    @OpenTraceabilityJsonAttribute(name = "gdst:productOwner")
    private PGLN productOwner;

    @Override
    public PGLN getProductOwner() {
        return this.productOwner;
    }


    @OpenTraceabilityAttribute(ns = Constants.GDST_NAMESPACE, name = "humanWelfarePolicy")
    @OpenTraceabilityJsonAttribute(name = "gdst:humanWelfarePolicy")
    private String humanWelfarePolicy;

    @OpenTraceabilityJsonAttribute(name = "cbvmda:unloadingPort")
    @OpenTraceabilityAttribute(ns = Constants.CBVMDA_NAMESPACE, name = "unloadingPort")
    private String unloadingPort;

    // Add constructor, getters, and setters here
    public GDSTTransshipmentEvent()
    {
        this.businessStep = URI.create("urn:gdst:bizStep:transshipment");
        this.action = EventAction.OBSERVE;
    }
}

File: CSharp/OpenTraceability.MSC/Events/Gdst12/GDSTTransshipmentEvent.cs
using OpenTraceability.GDST.Events;
using OpenTraceability.Models.Events;
using OpenTraceability.Models.Identifiers;
using OpenTraceability.Utility.Attributes;

namespace OpenTraceability.MSC.Events
{
    /// <summary>
    /// A transshipment event as defined by GDST 1.2: OBJECT / OBSERVE / business step "urn:gdst:bizStep:transshipment".
    /// GDST 2.0 models a transshipment as a shipping and a receiving event instead, so this single event has no direct
    /// 2.0 equivalent. It keeps productOwner and humanWelfarePolicy, which GDST 1.2 carried on the event itself.
    /// </summary>
    public class GDSTTransshipmentEvent : ObjectEvent<GDSTILMD>, IGDSTILMDEvent
    {
        [OpenTraceability(Constants.GDST_NAMESPACE, "productOwner")]
        [OpenTraceabilityJson("gdst:productOwner")]
        public PGLN? ProductOwner { get; set; }

        [OpenTraceability(Constants.GDST_NAMESPACE, "humanWelfarePolicy")]
        [OpenTraceabilityJson("gdst:humanWelfarePolicy")]
        public string? HumanWelfarePolicy { get; set; }

        [OpenTraceability(Constants.CBVMDA_NAMESPACE, "unloadingPort")]
        [OpenTraceabilityJson("cbvmda:unloadingPort")]
        public string? UnloadingPort { get; set; }

        public GDSTTransshipmentEvent()
        {
            this.BusinessStep = new Uri("urn:gdst:bizStep:transshipment");
            this.Action = EventAction.OBSERVE;
        }
    }
}
